File: CS/XtraTabWithButtons/Tab/Page/XtraTabPageCollectionDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraTab;

namespace XtraTabWithButtons
{
    public class XtraTabPageCollectionDescendant : XtraTabPageCollection
    {
        public XtraTabPageCollectionDescendant(XtraTabControl tabControl)
            : base(tabControl) { }

        protected override XtraTabPage CreatePage()
        {
            return new XtraTabPageDescendant();
        }
    }
}

File: VB/XtraTabWithButtons/Tab/Page/XtraTabPageCollectionDescendant.vb
Imports DevExpress.XtraTab

Namespace XtraTabWithButtons

    Public Class XtraTabPageCollectionDescendant
        Inherits XtraTabPageCollection

        Public Sub New(ByVal tabControl As XtraTabControl)
            MyBase.New(tabControl)
        End Sub

        Protected Overrides Function CreatePage() As XtraTabPage
            Return New XtraTabPageDescendant()
        End Function
    End Class
End Namespace
